dataactbroker/handlers/fileHandler.py

```diff
@@ -1225,6 +1171,44 @@ def get_d_file_url(task_key, file_type_name, cgac_code, start_date, end_date):
         url = CONFIG_BROKER["".join([file_type_name, "_url"])].format(cgac_code, start_date, end_date, callback)
         return url
 
+    @staticmethod
+    def create_generation_task(job_id):
+        sess = GlobalDB.db().session
+        task_key = uuid4()
+        task = FileGenerationTask(generation_task_key=task_key, job_id=job_id)
+        sess.add(task)
+        sess.commit()
+        return task.generation_task_key
+
+    def add_generation_job_info(self, file_type_name, job=None, dates=None):
+        # if job is None, that means the info being added is for detached d file generation
+        sess = GlobalDB.db().session
+        user_id = LoginSession.getName(session)
+
+        timestamped_name = s3UrlHandler.getTimestampedFilename(
+            CONFIG_BROKER["".join([str(file_type_name), "_file_name"])])
+        if self.isLocal:
+            upload_file_name = "".join([CONFIG_BROKER['broker_files'], timestamped_name])
+        else:
+            upload_file_name = "".join([str(user_id), "/", timestamped_name])
+
+        if job is None:
+            new_job = Job(filename=upload_file_name, original_filename=timestamped_name,
+                          job_status_id=JOB_STATUS_DICT['running'], job_type_id=JOB_TYPE_DICT['file_upload'],
+                          user_id=user_id, file_type_id=FILE_TYPE_DICT[file_type_name], start_date=dates['start_date'],
+                          end_date=dates['end_date'])
+            sess.add(new_job)
+            sess.commit()
+            return new_job
+
+        # This will update the reference so no need to return the job, just the upload and timestamped file names
+        job.filename = upload_file_name
+        job.original_filename = timestamped_name
+        job.job_status_id = JOB_STATUS_DICT["running"]
+        sess.commit()
+
+        return upload_file_name, timestamped_name
```

Comment:
This function returns two different types, which will make working with it confusing. Could it just always return the `job` (either the original or the new one)?

Relatedly, could we remove the `filename`, `original_filename`, `job_status_id` fields from the `Job` construction? Then we could pass both through the same code path:
```python
if job is None:
    job = Job(job_type_id=...)

job.filename = upload_file_name
...
sess.add(job)
sess.commit()

return job
```

We could always grab those fields from the created job.